lib/grpcServer/handlers/core/getBlockHandlerFactory.js

```diff
@@ -33,17 +33,26 @@ function getBlockHandlerFactory(insightAPI) {
     }
 
     let serializedBlock;
-
-    if (hash) {
-      serializedBlock = await insightAPI.getRawBlockByHash(hash);
-    } else {
-      serializedBlock = await insightAPI.getRawBlockByHeight(height);
+    let blockData;
+
+    try {
+      if (hash) {
+        serializedBlock = await insightAPI.getRawBlockByHash(hash);
```

Comment:
We should handle errors (buy response code) separately and throw InvalidArgumentGrpcError for each case if data not found.